# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/base/TextRange.cpp
# Commit: 5e92b9c7d479
# Full Hash: 5e92b9c7d4794af1d7ab0fb5c278c2e681a18fc6
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2023-01-28 21:11:06
# Regressor Bug: 1811092
# File Overlap Count: 1
# Description:
#   Bug 1811092 - P1: Add new async add/set text selection message. r=Jamie
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D167201
# ==============================================================================

diff -r 7cebaa85f4ff -r 5e92b9c7d479 accessible/base/TextRange.cpp
--- a/accessible/base/TextRange.cpp	Fri Jan 27 20:03:06 2023 +0000
+++ b/accessible/base/TextRange.cpp	Fri Jan 27 20:15:11 2023 +0000
@@ -263,51 +263,6 @@
   return true;
 }
 
-bool TextRange::SetSelectionAt(int32_t aSelectionNum) const {
-  HyperTextAccessible* root = mRoot->AsLocal()->AsHyperText();
-  if (!root) {
-    MOZ_ASSERT_UNREACHABLE("Not supported for RemoteAccessible");
-    return false;
-  }
-  RefPtr<dom::Selection> domSel = root->DOMSelection();
-  if (!domSel) {
-    return false;
-  }
-
-  RefPtr<nsRange> range = nsRange::Create(root->GetContent());
-  uint32_t rangeCount = domSel->RangeCount();
-  if (aSelectionNum == static_cast<int32_t>(rangeCount)) {
-    range = nsRange::Create(root->GetContent());
-  } else {
-    range = domSel->GetRangeAt(AssertedCast<uint32_t>(aSelectionNum));
-  }
-
-  if (!range) {
-    return false;
-  }
-
-  bool reversed;
-  AssignDOMRange(range, &reversed);
-
-  // If this is not a new range, notify selection listeners that the existing
-  // selection range has changed. Otherwise, just add the new range.
-  if (aSelectionNum != static_cast<int32_t>(rangeCount)) {
-    domSel->RemoveRangeAndUnselectFramesAndNotifyListeners(*range,
-                                                           IgnoreErrors());
-  }
-
-  IgnoredErrorResult err;
-  domSel->AddRangeAndSelectFramesAndNotifyListeners(*range, err);
-  if (!err.Failed()) {
-    // Changing the direction of the selection assures that the caret
-    // will be at the logical end of the selection.
-    domSel->SetDirection(reversed ? eDirPrevious : eDirNext);
-    return true;
-  }
-
-  return false;
-}
-
 void TextRange::ScrollIntoView(uint32_t aScrollType) const {
   LocalAccessible* root = mRoot->AsLocal();
   if (!root) {